dynamiqs/mesolve/mesolve.py

```diff
@@ -133,23 +134,17 @@ def _vectorized_mesolve(
 ) -> MEResult:
     # === vectorize function
     # we vectorize over H, jump_ops and rho0, all other arguments are not vectorized
-    # `n_batch` is a pytree. Each leaf of this pytree gives the number of times
-    # this leaf should be vmapped on.
-
-    # the result is vectorized over `_saved` and `infos`
-    out_axes = MEResult(None, None, None, None, 0, 0)
-
     if not options.cartesian_batching:
         broadcast_shape = jnp.broadcast_shapes(
             H.shape[:-2], rho0.shape[:-2], *[jump_op.shape[:-2] for jump_op in jump_ops]
         )
+        rho0 = jnp.broadcast_to(rho0, broadcast_shape + rho0.shape[-2:])
```

Comment:
```suggestion
        broadcast_shape = jnp.broadcast_shapes(H.shape, rho0.shape, *[L.shape for L in jump_ops])
        rho0 = jnp.broadcast_to(rho0, broadcast_shape)
```
Same for sesolve.

Reply:
Why do we need to broadcast the initial state?

Reply:
If your Hamiltonian is (k,n,n) for instance, your state must be (k,n,n) too. Otherwise you cannot store the information of all situations. You can extend this reasoning for additional dimensions.

Reply:
But it's automatic with broadcasting, i.e. the first time you do H @ rho you'll get a result with shape (k, n, n).

Reply:
Ha but then maybe JAX/diffrax is happy that rho keeps the same shape all along?

Reply:
After second though your comment won't hold for sesolve because it would broadcast a bra (n,1) into a dm (n,n)

Reply:
Because broadcsasting is from the left, so in the case of sesolve it would be (k, n, 1).

Reply:
No, it would be `(k, n, 1)`:
```python
H = jnp.ones((5, 2, 2))
rho = jnp.ones((2, 2))
print((H @ rho).shape)  # (5, 2, 2)
psi = jnp.ones((2, 1))
print((H @ psi).shape)  # (5, 2, 1)
```

Reply:
While I do agree in theory, in practise tests break when I remove these lines.

Reply:
I don't have time to investigate, and it's pretty minor, so I'd be tempted to keep it like that and to simply track it in the Linear. Unless @pierreguilmin you have an idea?